src/main/java/org/apache/sling/distribution/agent/impl/SimpleDistributionAgentQueueProcessor.java

```diff
@@ -173,6 +176,16 @@ private boolean processQueueItem(String queueName, DistributionQueueEntry queueE
         return removeItemFromQueue;
     }
 
+    private void addRandomDelay(){
+        int random = (int)(30 * Math.random() + 0);
```

Comment:
I think we need to penalize retries progressively, but in a probabilistic manner - so I'd propose:
* either, `rand(0, min(retry_count, 30)]` (pick a random number between 0 and `retry_count` to decide how long to wait)
* or, `rand[min(max(0, retry_count), 30), 30]` (pick a random number between `retry_count` and 30)

the latter has the characteristic that after 30 reattempts the wait will be a constant 30s (and thus different clients pointing to the same service may start to work in lock-step), whereas the former will always have variance.

The latter penalizes strongly on retries, the former is more lenient.

Reply:
@actinium15 I have updated my PR to implement the first approach suggested by you.